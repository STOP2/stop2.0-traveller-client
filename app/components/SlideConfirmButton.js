import React, { Component } from 'react'
import { View } from 'react-native'
import { SlideButton } from 'react-native-slide-button'
import { BoldText } from '../components/textComponents'

import styles from '../styles/stylesheet'

class SlideConfirmButton extends Component {
    render()
    {
        return (
        <View style={styles.sliderBackgroundRed} accessibilityComponentType="button" accessibilityLabel={this.props.text}>
          <SlideButton width={500} height={80}
           onSlideSuccess={this.props.onSlideSuccess}>
            <View style={styles.sliderTextContainer}>
              <BoldText style={styles.sliderText}>{this.props.text}</BoldText>
            </View>
          </SlideButton>
        </View>)
    }
}

SlideConfirmButton.propTypes = {
    onSlideSuccess: React.PropTypes.func.isRequired,
    text: React.PropTypes.string.isRequired
}

export default SlideConfirmButton
