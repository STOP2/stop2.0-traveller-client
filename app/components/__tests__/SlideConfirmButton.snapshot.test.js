import 'react-native'
import React from 'react'
import SlideConfirmButton from '../SlideConfirmButton'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'


jest.mock('react-native-localization')

describe('SlideConfirmButton component', () =>
{
    it('renders correctly', () =>
    {
        let tree = renderer.create(<SlideConfirmButton text="button" onSlideSuccess={() => {return null}} />).toJSON()

        expect(tree).toMatchSnapshot()
    })
})
