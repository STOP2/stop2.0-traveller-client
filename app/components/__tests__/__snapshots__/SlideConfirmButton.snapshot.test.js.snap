exports[`SlideConfirmButton component renders correctly 1`] = `
<View
  accessibilityComponentType="button"
  accessibilityLabel="button"
  style={
    Object {
      "backgroundColor": "#DC0451",
      "height": 80,
      "width": undefined,
    }
  }>
  <View
    style={
      Object {
        "height": 80,
        "overflow": "hidden",
        "width": 500,
      }
    }>
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "position": "relative",
        }
      }>
      <View
        style={
          Array [
            Object {
              "position": "absolute",
            },
            undefined,
            Object {
              "left": 0,
            },
          ]
        }>
        <View
          onLayout={[Function]}>
          <View
            style={
              Object {
                "height": 80,
                "width": 375,
              }
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": "gotham-rounded-medium",
                  },
                  Object {
                    "color": "white",
                    "fontSize": 25,
                    "margin": 20,
                    "position": "absolute",
                    "textAlign": "center",
                  },
                ]
              }>
              button
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
